homeassistant/components/azure_event_hub/config_flow.py

```diff
@@ -0,0 +1,208 @@
+        if user_input is None:
+            return self.async_show_form(step_id=STEP_USER, data_schema=BASE_SCHEMA)
+
+        self._conn_string = user_input.pop(CONF_USE_CONN_STRING)
+        self._data = user_input
+
+        if self._conn_string:
+            return await self.async_step_conn_string()
+        return await self.async_step_sas()
+
+    async def async_step_conn_string(
+        self, user_input: dict[str, Any] = None
+    ) -> FlowResult:
+        """Handle the connection string steps."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id=STEP_CONN_STRING,
+                data_schema=CONN_STRING_SCHEMA,
+                description_placeholders=self._data[CONF_EVENT_HUB_INSTANCE_NAME],
+                last_step=True,
+            )
+        errors = await self.async_update_and_validate_data(user_input)
+        if errors is not None:
+            return self.async_show_form(
+                step_id=STEP_CONN_STRING,
+                data_schema=CONN_STRING_SCHEMA,
+                errors=errors,
+                description_placeholders=self._data[CONF_EVENT_HUB_INSTANCE_NAME],
+                last_step=True,
+            )
+
+        return self.async_create_entry(
+            title=self._data[CONF_EVENT_HUB_INSTANCE_NAME],
+            data=self._data,
+            options=self._options,
+        )
+
+    async def async_step_sas(self, user_input: dict[str, Any] = None) -> FlowResult:
+        """Handle the sas steps."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id=STEP_SAS,
+                data_schema=SAS_SCHEMA,
+                description_placeholders=self._data[CONF_EVENT_HUB_INSTANCE_NAME],
+                last_step=True,
+            )
+
+        errors = await self.async_update_and_validate_data(user_input)
+        if errors is not None:
+            return self.async_show_form(
+                step_id=STEP_SAS,
+                data_schema=SAS_SCHEMA,
+                errors=errors,
+                description_placeholders=self._data[CONF_EVENT_HUB_INSTANCE_NAME],
+                last_step=True,
+            )
+
+        return self.async_create_entry(
+            title=self._data[CONF_EVENT_HUB_INSTANCE_NAME],
+            data=self._data,
+            options=self._options,
+        )
+
+    async def async_step_import(self, import_config: dict[str, Any]) -> FlowResult:
+        """Import config from configuration.yaml."""
+        if self._async_current_entries():
+            return self.async_abort(reason="single_instance_allowed")
+        if CONF_SEND_INTERVAL in import_config:
+            self._options[CONF_SEND_INTERVAL] = import_config.pop(CONF_SEND_INTERVAL)
+        if CONF_MAX_DELAY in import_config:
+            self._options[CONF_MAX_DELAY] = import_config.pop(CONF_MAX_DELAY)
+        self._data = import_config
+        return self.async_create_entry(
```

Comment:
We want to validate the connection before creating the entry also for imported data.

Reply:
Good one, I decided to abort in that case, is that the expected behaviour? Or should it start a flow at the right step?

Reply:
It's ok to abort in that case.